hysds/dataset_ingest.py

```diff
@@ -75,7 +75,8 @@ def index_dataset(grq_update_url, update_json):
     r = requests.post(
         grq_update_url, verify=False, data={"dataset_info": json.dumps(update_json)}
     )
-    r.raise_for_status()
+    if not 200 <= r.status_code < 300:
```

Comment:
@pymonger 
According to `r.raise_for_status()` it mostly checks if the status code is >= 400 [source code](https://requests.readthedocs.io/en/latest/_modules/requests/models/#Response.raise_for_status)
we don't need to check for anything in the 300 range b/c those are redirects

grq2's rest API doesnt do any redirects, it only returns 200's for success and 400/500's for errors

we can make backoff alittle more general and use a tuple of exceptions instead of checking for only a single exception type: 
https://github.com/litl/backoff/blob/64de65d56a2352ca093eb1906dcc471d058aa214/backoff/_decorator.py#L143-L144

```python
@backoff.on_exception(
    backoff.expo,
    (NotAllProductsIngested, requests.RequestException),
    max_tries=backoff_max_tries,
    max_value=backoff_max_value,
)
```

Reply:
Thanks @DustinKLo. I never realized what `raise_for_status()` was explicitly looking for. 
Yes, please go ahead and pass in a tuple of those exceptions to backoff.